jujugui/static/gui/src/app/models/search-results-extension.js

```diff
@@ -0,0 +1,84 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('search-result-extension', function(Y) {
+  var ns = Y.namespace('juju.models'),
+      utils = Y.namespace('juju.views.utils');
+
+  /**
+   * Class extension that allows models to be displayed as search results.
+   *
+   * @namespace juju.models
+   * @class SearchResult
+   */
+  function SearchResult() {}
+
+  SearchResult.prototype = {
+
+    /**
+      Parse the owner from the ID.
+      @method ownerFromId
+      @return {String} the charm owner.
+     */
+    ownerFromId: function() {
+      var id = this.get('id'),
+        owner = id.split('/')[0];
+      if (owner.indexOf('~') === 0) {
+        return owner.replace('~', '');
+      } else {
+        return 'charmers';
+      }
+    },
+
+    /**
+      Produces a search results object for rendering in the search result templates.
+      @method toSearchResult
+      @param {Object} charmstore the charmstore being searched.
+      @return {Object} a plain Javascript object containing bundle attributes.
+    */
+    toSearchResult: function(charmstore) {
+      var type = this.get('entityType');
```

Comment:
instead of making all of these 'get' calls, it's more efficient to `this.getAttrs()` then deal with a native object.